lib/types/src/memory_view.rs

```diff
@@ -64,6 +64,33 @@ where
             _phantom: PhantomData,
         }
     }
+
+    /// Creates a subarray view from this MemoryView.
+    pub fn subarray(&self, start: u32, end: u32) -> Self {
+        assert!(start <= end);
+        assert!((end as usize) < self.length);
+        Self {
+            ptr: unsafe { self.ptr.add(start as usize) },
```

Comment:
ptr.add accounts for size, does the bounds checking code above account for size? If length is already in terms of `T` and not `u8` then this should be good 👍

Reply:
Yeah, length is in terms of `T`.